File: src/C3.Catalogue/CassetteModels/ICassetteModelRepository.vb
Namespace CassetteModels

    Public Interface ICassetteModelRepository

        Function GetAll() As IList(Of CassetteModel)
        Function FindByIdentifier(identifier As String) As CassetteModel
        Function BrandExists(code As String) As Boolean
        Function IdentifierExists(identifier As String) As Boolean
        Function IsReferencedByTape(identifier As String) As Boolean
        Sub Add(value As CassetteModel)
        Sub Update(value As CassetteModel)
        Sub Delete(identifier As String)

    End Interface

End Namespace

File: src/C3.Catalogue/CassetteModels/ICassetteModelRepository.cs
using System.Collections.Generic;

namespace C3.Catalogue.CassetteModels
{
    public interface ICassetteModelRepository
    {
        IList<CassetteModel> GetAll();

        CassetteModel FindByIdentifier(string identifier);

        bool BrandExists(string code);

        bool IdentifierExists(string identifier);

        bool IsReferencedByTape(string identifier);

        void Add(CassetteModel value);

        void Update(CassetteModel value);

        void Delete(string identifier);
    }
}
